pkg/resources/envoy/configmap.go

```diff
@@ -327,31 +375,95 @@ func GenerateEnvoyConfig(kc *v1beta1.KafkaCluster, elistener v1beta1.ExternalLis
 		log.Error(err, "could not marshall envoy tcp_proxy config")
 		return ""
 	}
-	listeners = append(listeners, &envoylistener.Listener{
-		Address: &envoycore.Address{
-			Address: &envoycore.Address_SocketAddress{
-				SocketAddress: &envoycore.SocketAddress{
-					Address: "0.0.0.0",
-					PortSpecifier: &envoycore.SocketAddress_PortValue{
-						PortValue: uint32(elistener.GetAnyCastPort()),
+
+	// Create an any cast broker access point
+	if elistener.ExternalStartingPort == -1 {
+		// append l[AnyCastPort] with another filter for the same listener
+		filterChain = &envoylistener.FilterChain{
```

Comment:
Could we avoid somehow code duplication here? I see that only the hostname is different from the broker specific filter.